docs: Add incremental replication example (#1390)
meltano/sdk: docs/incremental_replication.md

## Example Code: Timestamp-Based Incremental Replication

```py
class CommentsStream(RESTStream):

    replication_key = "date_gmt"
    is_sorted = True

    schema = th.PropertiesList(
        th.Property("date_gmt", th.DateTimeType, description="date"),
    ).to_dict()

    def get_url_params(
        self, context: dict | None, next_page_token
    ) -> dict[str, Any]:
        params = {}
        if starting_date := self.get_starting_timestamp(context):
            params["after"] = starting_date.isoformat()
        if next_page_token is not None:
            params["page"] = next_page_token
        self.logger.info(f"QUERY PARAMS: {params}")
        return params

    url_base = "https://example.com/wp-json/wp/v2/comments"
    authenticator = None
```

> First we inform the SDK of the `replication_key`, which automatically triggers incremental import mode. Second, optionally, set `is_sorted` to true; with this setting, Singer will throw an error if a supposedly incremental import sends results older than the starting timestamp.

Reviewer: This is still unclear to me how this property works. Could there be a concrete example provided?

Reviewer: I just realized my comment is confusing, what is unclear to me is how the `is_sorted` property is used and I think an example might help.

Like what if the property is set to True and some data points come in, what happens exactly in the messages that are sent to the target? And what if the same data points come in, but the property was set to False, what would be different? Something along those lines.

> Last, we have to adapt the query to the remote system, in this example by adding a query parameter with the ISO timestamp.

Reviewer 2: ```suggestion
3. Last, we have to adapt the query to the remote system, in this example by adding a query parameter with the ISO timestamp.
```

Note:
- unlike a `primary_key`, a `replication_key` does not have to be unique

> - in incremental replication, it is OK to resend rows where the replication key is equal to previous highest key.

Reviewer: I would add here that it is ok to resend rows because the target will just update the row with the same values instead of inserting duplicates. At least that's the behavior of the target [snowflake](https://hub.meltano.com/loaders/target-snowflake/) I'm familiar with.

> Note:
> - unlike a `primary_key`, a `replication_key` does not have to be unique
> - in incremental replication, it is OK to resend rows where the replication key is equal to previous highest key.

Reviewer 2: ```suggestion

```{note}
- The SDK will throw an error if records come out of order when `is_sorted` is true.
- Unlike a `primary_key`, a `replication_key` does not have to be unique
- In incremental replication, it is OK and usually recommended to resend rows where the replication key is equal to previous highest key.
```

## Manually testing incremental import during development

> To test the tap stand-alone, manually create a state file and run the tap:

Reviewer 2: ```suggestion
To test the tap in standalone mode, manually create a state file and run the tap:
```
